src/main/java/org/opensearch/ad/task/ADTaskManager.java

```diff
@@ -1099,6 +1150,23 @@ private void resetHistoricalDetectorTaskState(ADTask adTask, AnomalyDetectorFunc
         }));
     }
 
+    private boolean isTaskStopped(String taskId, AnomalyDetector detector, ADTaskProfile taskProfile) {
+        boolean taskStopped = false;
+        String detectorId = detector.getDetectorId();
+        if (taskProfile == null || !Objects.equals(taskId, taskProfile.getTaskId())) {
+            logger.debug("AD task not found for task {} detector {}", taskId, detectorId);
+            // If no node is running this task, reset it as STOPPED.
+            taskStopped = true;
+        } else if (!detector.isMultientityDetector() && taskProfile.getNodeId() == null) {
+            logger.debug("AD task not running for single flow detector {}, task {}", detectorId, taskId);
```

Comment:
single entity or stream to be consistent?

Reply:
sure, will change to single entity